zzk/service/host.go

```diff
@@ -101,6 +103,8 @@ func (l *HostStateListener) GetPath(nodes ...string) string {
 
 // Ready adds an ephemeral node to the host registry
 func (l *HostStateListener) Ready() error {
+	l.nodelock.Lock()
```

Comment:
Since you are locking access to this node, do you want to lock the Done method as well?  What about the Read at 162?